ShellCheck/Analytics.hs

```diff
@@ -2548,7 +2550,47 @@ checkUncheckedCd params root =
                 warn (getId t) 2164 "Use 'cd ... || exit' or 'cd ... || return' in case cd fails."
```

Comment:
Neat! How about adding pushd/popd to the checkUncheckedCd method to avoid reiterating the tree several times. Maybe something like 

    checkElement t@T_SimpleCommand {} =
      when (name `elem` ["cd", "pushd", "popd"])
          ...  $ do
          warn (getId t) 2164 $ "Use '" ++ name ++ " ... || exit'  ...."
      where
        name = fromMaybe "" $ getCommandName t
        isSafeDir t = case oversimplify t of 
              [_, ".."] -> True; 
              _ -> False